lib/msf/core/exploit/remote/http/splunk/helpers.rb

```diff
@@ -0,0 +1,98 @@
+# -*- coding: binary -*-
+
+# Module with helper methods for other Splunk module methods
+module Msf::Exploit::Remote::HTTP::Splunk::Helpers
+  # Helper methods are private and should not be called by modules
+
+  private
+
+  # Helper method to get tokens for login
+  #
+  # @param timeout [Integer] The maximum number of seconds to wait before the request times out
+  # @return [String, nil] Post data to use for login
+  def splunk_helper_extract_token(timeout = 20)
+    res = send_request_cgi({
+      'uri' => splunk_url_login,
+      'method' => 'GET',
+      'keep_cookies' => true
+    }, timeout)
+
+    unless res&.code == 200
+      vprint_error('Unable to get login tokens')
+      return nil
+    end
+    "session_id_8000=#{rand_text_numeric(40)}; " << res.get_cookies
```

Comment:
Is this 8000 hardcoded to the current port number, if so do we need to handle the scenario when it isn't?

Note for the tester. Test this against another port.

Reply:
I changed `session_id_8000` to `session_id_8001` and it'd still login for splunk running on port `8000`, however it looks like it does refer to the port number as seen [here](https://github.com/rapid7/metasploit-framework/blob/2390058b375decd201edcbcd6b33db24f599a16f/modules/auxiliary/scanner/http/splunk_web_login.rb#L78), so I'll make it `session_id_#{datastore['RPORT']}`